src/core/service.c

```diff
@@ -202,14 +202,24 @@ static void service_stop_watchdog(Service *s) {
 
 static void service_start_watchdog(Service *s) {
         int r;
+        usec_t watchdog_usec;
 
         assert(s);
 
-        if (s->watchdog_usec <= 0)
+        if (s->watchdog_usec == 0 || s->watchdog_usec == USEC_INFINITY)
                 return;
 
+        if (s->watchdog_override_usec == USEC_INFINITY)
+                 return;
+
+        if (s->watchdog_override_usec != 0)
+            watchdog_usec = s->watchdog_override_usec;
+        else
+            watchdog_usec = s->watchdog_usec;
```

Comment:
Indentation is borked. Please see CODING_STYLE: 8ch indentation please!

Also, could you please create a new function for this: service_get_watchdog_usec() which returns the current watchdog interval, either taking the override value or the original value?